src/shim/kernel.rs

```diff
@@ -96,7 +70,10 @@ error_number! {
 
 impl fmt::Display for ErrorNumber {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.as_unshimmed().fmt(f)
+        match self {
+            Self::Unknown(u) => write!(f, "{}", u),
+            e => write!(f, "{}", e),
+        }
```

Comment:
Are you sure this works? Looks like an infinite loop to me.

Reply:
Testing it.

Reply:
Fixing it.